lib/Basics/HybridLogicalClock.h

```diff
@@ -107,7 +107,7 @@ class HybridLogicalClock {
   static std::string encodeTimeStamp(uint64_t t) {
     std::string r(11, '\x00');
     size_t pos = 11;
-    while (t > 0) {
+    while (pos > 0) {
```

Comment:
Minor change:

after that loop variable `pos` will be always 0 and the below statement:
`return r.substr(pos, 11 - pos);` is equal to `return r`
What do you think if we change below code:
```
r[--pos] = encodeTable[static_cast<uint8_t>(t & 0x3ful)]; 
```
to
```
r[--pos] = encodeTable[static_cast<uint8_t>(t & 0x3ful)];
if (t == 0) {
    break
}
```

so if variable `t` first time hit the zero then it writes `-` at the beginning of string, break the loop and then the below return makes sense. When t==0 then the string will be `-` instead of `-----------`